repos/c2sm/packages/icon/package.py

```diff
@@ -522,11 +530,12 @@ def configure_args(self):
                 config_vars['CLAWFLAGS'].append(
                     self.spec['libcdi-pio'].headers.include_flags)
 
-        if '~cuda' in self.spec:
+        gpu = self.spec.variants['gpu'].value
+        if gpu == 'no':
             config_args.append('--disable-gpu')
-        else:
+        elif gpu == 'openacc+cuda':
```

Comment:
```suggestion
        else:
```
There are only two legal values for `gpu`, so there's no other need to test the other case.